module/zcommon/zfeature_common.c

```diff
@@ -217,8 +217,11 @@ zfs_mod_supported_feature(const char *name)
 	 * libzpool, always supports all the features. libzfs needs to
 	 * query the running module, via sysfs, to determine which
 	 * features are supported.
+	 *
+	 * The equivalent _can_ be done on FreeBSD by way of the sysctl
+	 * tree, but this has not been done yet.
 	 */
-#if defined(_KERNEL) || defined(LIB_ZPOOL_BUILD)
+#if defined(_KERNEL) || defined(LIB_ZPOOL_BUILD) || defined(__FreeBSD__)
 	return (B_TRUE);
 #else
 	return (zfs_mod_supported(ZFS_SYSFS_POOL_FEATURES, name));
```

Comment:
Just a thought, and definitely not for now, but maybe in the future, this could be redone as a new shiny ZFS_IOC, instead of all platforms finding their own way to query features? Almost a shame we didn't do that with `zpool version` and it could be extended version information.

Reply:
We definitely considered it at the time, but using sysfs on Linux afforded us some other benefits.  For example, it made it easy for scripts or other simple utilities to check if a feature or property was understood by the kernel.  Using an ioctl would definitely be more probable which would be great, perhaps as part of the FreeBSD or MacOS work?  The two approaches aren't mutually exclusive so I don't see why we couldn't have both.

Reply:
Yeah, with `zpool version` I just look it up with `sysctl` - it was easy after all :)